rosetta-lang/src/main/java/com/regnosys/rosetta/formatting2/RuneFormatter.java

```diff
@@ -0,0 +1,82 @@
+package com.regnosys.rosetta.formatting2;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.formatting2.FormatterRequest;
+import org.eclipse.xtext.formatting2.regionaccess.ITextRegionAccess;
+import org.eclipse.xtext.formatting2.regionaccess.ITextReplacement;
+import org.eclipse.xtext.formatting2.regionaccess.TextRegionAccessBuilder;
+import org.eclipse.xtext.formatting2.regionaccess.internal.TextRegionRewriter;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.util.TextRegion;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+public class RuneFormatter implements CodeFormatterService{
+	@Inject
+	private Provider<FormatterRequest> formatterRequestProvider;
+	
+	@Inject 
+	private Provider<RosettaFormatter> rosettaFormatterProvider;
+
+	@Inject
+	private TextRegionAccessBuilder regionBuilder;
+
+	@Override
+	public List<Resource> formatCollection(List<Resource> resources) throws IOException {
+		
+		List<Resource> result = new ArrayList<Resource>();
+		
+		for (Resource doc: resources) {
+			Resource formattedDocument = formatDocument(doc);
+			result.add(formattedDocument);
+		}	
+		return result;
```

Comment:
This is perfectly fine, but just as a challenge, could you rewrite this using Java stream API? :)